src/tools/illink/src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -364,12 +379,15 @@ static void VerifyDamOnPropertyAndAccessorMatch (SymbolAnalysisContext context,
 				// None on parameter of 'set' matches unannotated
 				|| methodSymbol.MethodKind == MethodKind.PropertySet
 				&& methodSymbol.Parameters[methodSymbol.Parameters.Length - 1].GetDynamicallyAccessedMemberTypes () != DynamicallyAccessedMemberTypes.None) {
-				context.ReportDiagnostic (Diagnostic.Create (
-					DiagnosticDescriptors.GetDiagnosticDescriptor (DiagnosticId.DynamicallyAccessedMembersConflictsBetweenPropertyAndAccessor),
-					methodSymbol.AssociatedSymbol!.Locations[0],
-					methodSymbol.AssociatedSymbol!.GetDisplayName (),
-					methodSymbol.GetDisplayName ()
-				));
+				var associatedSymbol = methodSymbol.AssociatedSymbol!;
+				if (associatedSymbol.Locations.Length > 0) {
+					context.ReportDiagnostic (Diagnostic.Create (
+						DiagnosticDescriptors.GetDiagnosticDescriptor (DiagnosticId.DynamicallyAccessedMembersConflictsBetweenPropertyAndAccessor),
+						associatedSymbol.Locations[0],
```

Comment:
The diagnostics doesn't seem to require location (it's explicitly annotated as nullable), so I think a better solution would be to just pass in null if we don't have it.

Reply:
There's also a singleton Location.None